common/scripted_effects/wc_artifacts_generic_scripted_effects.txt

```diff
@@ -735,248 +735,3119 @@ give_new_artifact_random_name_and_origin_effect = {
 ###   Spawn Effects   ###
 #########################
 
-#########################
-### Gamestart Effects ###
-#########################
+# Use this before effects when character supposed to get only these artifacts they can use
+# set_variable = { which = local_new_generic_artifact_receiving_character_matter_variable value = 1 }
+
+# This is used in gamestart effect ONLY
+# set_variable = { which = local_new_generic_artifact_gamestart_variable value = 1 }
+
+# You need to use this thing before spawning effects, X is amount of artifacts.
+# while = { count = X set_local_variables_for_new_received_generic_artifact_effect = yes }
+
+set_local_variables_for_new_received_generic_artifact_effect = {
+	if = { # Set rarity if not already set
+		limit = { check_variable = { which = local_new_received_generic_artifact_rarity_variable value == 0 } }
+		random_list = { # All items in 4 options will have the same rarity
+			1 = {
+				generic_artifact_rarity_legendary_score = yes
+				set_variable = { which = local_new_received_generic_artifact_rarity_variable value = 5 }
+			}
+			1 = {
+				generic_artifact_rarity_epic_score = yes
+				set_variable = { which = local_new_received_generic_artifact_rarity_variable value = 4 }
+			}
+			1 = {
+				generic_artifact_rarity_rare_score = yes
+				set_variable = { which = local_new_received_generic_artifact_rarity_variable value = 3 }
+			}
+			1 = {
+				generic_artifact_rarity_uncommon_score = yes
+				set_variable = { which = local_new_received_generic_artifact_rarity_variable value = 2 }
+			}
+			1 = {
+				generic_artifact_rarity_common_score = yes
+				set_variable = { which = local_new_received_generic_artifact_rarity_variable value = 1 }
+			}
+		}
+	}
+
+	change_variable = { which = local_new_received_generic_artifact_amount_variable value = 1 } # Count times you used this effect
 
-game_start_spawn_artifacts_random_effect = { # Slots -> Types
 	random_list = {
-		1 = { # Neck slot
-			trigger = { can_get_amulet_at_gamestart_distribution_trigger = yes }
-			random_list = {
-				10 = {
-					random_list = {
-						1 = { add_artifact = amulet_5_diplomacy }
-						2 = { add_artifact = amulet_4_diplomacy }
-						4 = { add_artifact = amulet_3_diplomacy }
-						8 = { add_artifact = amulet_2_diplomacy }
+		3 = { # Jewelry
+			trigger = {
+
+				check_variable = { which = local_new_received_generic_artifact_type_jewelry_amount_variable value < 3 } # Because 3 is max, since there is only 3 jewelry types
+
+				#trigger_if = {
+					#limit = { check_variable = { which = local_new_generic_artifact_receiving_character_matter_variable value = 1 } }
+					trigger_if = {
+						limit = { check_variable = { which = local_new_generic_artifact_gamestart_variable value = 1 } }
```

Comment:
This is supposed to be boolean, yes? Thus it has either 0 or 1?

Reply:
> This is supposed to be boolean, yes? Thus it has either 0 or 1?

Yeah.